ja_JP/readme.md

```diff
@@ -0,0 +1,738 @@
+test.skip('will not be run', t => {
+	t.fail();
+});
+```
+
+### Before & after hooks
+
+setup とteardownの両方かいずれかが必要なとき、`test()`と同じように、`test.before()`と`test.after()`を利用できます。`test.before()`と`test.after()`に与えられたテスト関数は、全てのテストの前/後に呼び出されます。それぞれのテストにsetup/teardownが必要なときは、`test.beforeEach()`と`test.afterEach()`が利用できます。必要なだけ追加してください。失敗したときに表示されるタイトルを特定することも必要に応じて出来ます。
+
+
+```js
+test.before(t => {
+	// 全てのテストの前に実行します
+});
+
+test.before(t => {
+	// テストの前に実行しますが、上記の処理の後です
+});
+
+test.after('cleanup', t => {
+	// 全てのテストの後に実行します
+});
+
+test.beforeEach(t => {
+	// それぞれのテストの前に実行します
+});
+
+test.afterEach(t => {
+	// それぞれのテストの後に実行します
+});
+
+test(t => {
+	// 通常のテスト
+});
+```
+
+これらのいずれにおいても、asyncを使うことも、asyncオブジェクトを返すことも、"コールバックモード"を有効にすることも、出来ます。
+
+```js
+test.before(async t => {
+	await promiseFn();
+});
+
+test.cb.beforeEach(t => {
+	setTimeout(t.end);
+});
+
+test.afterEach.cb(t => {
+	setTimeout(t.end);
+});
+
+test.after(t => {
+   return new Promise(/* ... */);
+});
+```
+
+`beforeEach`と`afterEach`はテストでコンテキストを共有できます:
+
+```js
+test.beforeEach(t => {
+	t.context.data = generateUniqueData();
+});
+
+test(t => {
+	t.is(t.context.data + 'bar', 'foobar');
+});
+```
+
+このコンテキストは、デフォルトでは1つのオブジェクトですが、直接アサインすることも出来ます。
+
+```js
+test.beforeEach(t => {
+	t.context = 'unicorn';
+});
+
+test(t => {
+	t.is(t.context, 'unicorn');
+});
+```
+
+### テスト修飾子のチェーン
+
+テストの修飾子を以下のようにチェーンすることができます。
+
+```js
+test.before.skip([title], testFn);
+test.skip.after(....);
+test.serial.only(...);
+test.only.serial(...);
+```
+
+これはテストで、他の修飾子の振る舞いや情報を失わないで、一時的に`skip`や`only`を使うときに、特に有用です。
+
+### カスタムのアサーションモジュール
+
+AVAに加えて、もしくは、代わりにどのようなアサーションモジュールを利用することもできますが、[今のところ](https://github.com/sindresorhus/ava/issues/25)`.plan()`メソッドを使うことはできません。
+
+```js
+import assert from 'assert';
+
+test(t => {
+	assert(true);
+});
+```
+
+### ES2015のサポート
+
+AVAは、[Babel 6](https://babeljs.io)によってES2015をサポートしています。ES2015でテストを書いてください。他の環境整備は必要ありません。プロジェクト内でバージョンを問わずにBabelを利用することができます。私たちは、[`es2015`](http://babeljs.io/docs/plugins/preset-es2015/) と[`stage-2`](http://babeljs.io/docs/plugins/preset-stage-2/)のプリセットで独自にバンドルしたBabelを使用しています。
+
+#### importしたモジュールのトランスパイル
+
+AVAは現時点では、実行するように依頼したテストだけをトランスパイルします。*テストの外から```import```したモジュールはトランスパイルしません。*このアプローチには妥当な理由がありますが、期待した挙動ではないかもしれません！
+
+シンプルな次善策として、続けてインポートされるモジュールのトランスパイルを行うために、[Babel's require hook](https://babeljs.io/docs/usage/require/)を利用できます。
+
+```js
+import test from 'ava';
+import 'babel-core/register';
+import foo from './foo'; // <-- ES2015 で書くことが出来ます！
+
+test('foo bar', t => {
+	t.same('baz', foo('bar'));
+});
+```
+
+[#111](https://github.com/sindresorhus/ava/issues/111)は、将来性のあるエンハンスメントとして、この問題を追いかけています。
+
+
+### Promiseのサポート
+
+テストの中でpromiseを返したら、テストはpromiseがresolveしたときに終了するので、明示的にテストを終了する必要はありません。
+
+```js
+test(t => {
+	return somePromise().then(result => {
+		t.is(result, 'unicorn');
+	});
+});
+```
+
+### generator関数のサポート
+
+AVAは[generator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*)をサポートしています。
```

Comment:
There is a link to mozilla in Japanese. It may be better ? (`https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Statements/function*`)